File: src/main/java/com/hrms/hr_payroll_management_system/service/audit/AuditLogService.java
package com.hrms.hr_payroll_management_system.service.audit;

import com.hrms.hr_payroll_management_system.dto.response.audit.AuditLogResponse;
import com.hrms.hr_payroll_management_system.enums.AuditAction;

import java.util.List;

public interface AuditLogService {

    void log(
            AuditAction action,
            String entityType,
            Long entityId,
            String description
    );

    List<AuditLogResponse> getAll();

    List<AuditLogResponse> getByActor(
            String actor
    );

    List<AuditLogResponse> getByEntity(
            String entityType,
            Long entityId
    );
}

File: src/services/auditLog.service.js
import { auditLogApi } from '../api/auditLog.api';

export const auditLogService = {
  async list() {
    const res = await auditLogApi.getAll();
    return res.data || [];
  },

  async listByActor(actor) {
    const res = await auditLogApi.getByActor(actor);
    return res.data || [];
  },

  async listByEntity(entityType, entityId) {
    const res = await auditLogApi.getByEntity(entityType, entityId);
    return res.data || [];
  },
};